src/transformers/trainer.py

```diff
@@ -506,14 +520,14 @@ def __init__(
             logger.info(f"Using {args.half_precision_backend} half precision backend")
 
         self.do_grad_scaling = False
-        if (args.fp16 or args.bf16) and not args.deepspeed:  # deepspeed manages its own half precision
+        if (args.fp16 or args.bf16) and not (
+            args.deepspeed or is_sagemaker_mp_enabled()
+        ):  # deepspeed and SageMaker Model Parallel manage their own half precision
```

Comment:
```suggestion
        if (args.fp16 or args.bf16) and not (args.deepspeed or is_sagemaker_mp_enabled()):
            # deepspeed and SageMaker Model Parallel manage their own half precision  
```
Better like this.

Reply:
Updated with the reformatting suggestion, thanks!